stdlib/buffer.gr

```diff
@@ -0,0 +1,680 @@
+/**
+ * @module Buffer: Utilities for working with buffers.
+ *
+ * @example import Buffer from "buffer"
+ *
+ * @since v0.4.0
+ */
+
+import Memory from "runtime/unsafe/memory"
+import WasmI32 from "runtime/unsafe/wasmi32"
+import Conv from "runtime/unsafe/conv"
+import Exception from "runtime/exception"
+import Int32 from "int32"
+import Bytes from "bytes"
+import String from "string"
+import { coerceNumberToWasmI32 } from "runtime/numbers"
+
+/**
+ * @section Types: Type declarations included in the Buffer module.
```

Comment:
Just a quick note—we might not want this section since the record type isn't exported.